lib/kuzzle/event/KuzzleEventEmitter.ts

```diff
@@ -183,11 +208,13 @@ class KuzzleEventEmitter extends EventEmitter {
     const events = getWildcardEvents(event);
     const funcs = [];
 
-    for (let i = 0; i < events.length; i++) {
```

Comment:
This old for loop was choose to increase performance, I don't know if `for (let i = 0; i < ...)` is still faster than `for (const ... of ...)` in later Node.js versions

Reply:
I did that change because it was a warning from Sonarqube, and after looking at some benchmarks, it takes almost the same time for the low iteration count we have. 
Here are the results for 10 elements which we probably won't even reach in most cases:
![image](https://github.com/user-attachments/assets/e6a6d102-89ee-4ec3-8a9a-6deb202f3a34)

Reply:
The related Sonar rule: https://rules.sonarsource.com/typescript/RSPEC-4138/